odoo/modules/registry.py

```diff
@@ -472,23 +475,29 @@ def check_indexes(self, cr, model_names):
                    [tuple(row[0] for row in expected)])
         existing = {row[0] for row in cr.fetchall()}
 
-        if not self.has_trigram and any(row[3] == 'trigram' for row in expected):
-            self.has_trigram = sql.install_pg_trgm(cr)
-
-        for indexname, tablename, columnname, index in expected:
+        for indexname, tablename, column_name, index, unaccent in expected:
             assert index in ('btree', 'btree_not_null', 'trigram', True, False, None)
             if index and indexname not in existing:
+                column_expression = f'"{column_name}"'
                 method = 'btree'
                 operator = ''
                 where = ''
                 if index == 'btree_not_null':
-                    where = f'"{columnname}" IS NOT NULL'
+                    where = f'{column_expression} IS NOT NULL'
                 elif index == 'trigram' and self.has_trigram:
                     method = 'gin'
                     operator = 'gin_trgm_ops'
+                    # Add `unaccent` to the trigram index only because they are mainly used for (i/=)like search
+                    # And unaccent is added only in these cases by the ORM.
```

Comment:
```suggestion
                    # add `unaccent` to the trigram index only because they are mainly used for (i/=)like search
                    # and unaccent is added only in these cases when searching
```